web/src/app/rotations/RotationForm.js

```diff
@@ -1,54 +1,81 @@
-import React from 'react'
+import React, { useEffect, useMemo, useState } from 'react'
 import p from 'prop-types'
 import { FormContainer, FormField } from '../forms'
 import { TimeZoneSelect } from '../selection'
-import { TextField, Grid, MenuItem } from '@material-ui/core'
+import {
+  TextField,
+  Grid,
+  MenuItem,
+  Typography,
+  makeStyles,
+  Switch,
+  FormControlLabel,
+} from '@material-ui/core'
 import { startCase } from 'lodash'
-import { DateTime, Info } from 'luxon'
-import { ISOTimePicker } from '../util/ISOPickers'
+import { ISODateTimePicker } from '../util/ISOPickers'
+import NumberField from '../util/NumberField'
+import { DateTime } from 'luxon'
+import { useQuery, gql } from '@apollo/client'
+import { getHours } from './util'
+
+const query = gql`
+  query upcomingHandoffTimes($input: UpcomingHandoffTimesInput) {
+    upcomingHandoffTimes(input: $input)
+  }
+`
 
 const rotationTypes = ['hourly', 'daily', 'weekly']
 
+const useStyles = makeStyles({
+  listNone: {
+    listStyle: 'none',
+  },
+  bolder: {
+    fontWeight: 'bolder',
+  },
+  flex: {
+    display: 'flex',
+  },
+})
+
 export default function RotationForm(props) {
-  function dayOfWeek() {
-    const { start, timeZone } = props.value
-    return DateTime.fromISO(start, { zone: timeZone }).weekday
-  }
+  const { value } = props
+  const classes = useStyles()
+  const localZone = useMemo(() => DateTime.local().zone.name, [])
+  const [configInZone, setConfigInZone] = useState(false)
+  const configZone = configInZone ? value.timeZone : 'local'
+  const [upcomingHandoffs, setUpcomingHandoffs] = useState([])
```

Comment:
Don't use state for data, just render the data from apollo

Reply:
`data || []` will produce the same result without using the state if it's undefined.

Which input field becomes invalid? Generally, the form value should always remain valid and an invalid input should be reset to "last correct value" on blur -- sounds like a potential bug. Otherwise, it would be better to store the "last valid input" in the state and use that to fetch the data as a source of truth than to try to cache something down the chain.

Reply:
Hmm. Maybe we can use placeholder text in the case the data is missing/invalid to solve for that. Showing the wrong times doesn't seem like the right answer either, even if it's temporary. We need to account for slow network and errors

There's also the issue of the flicker when the dialog first opens and we don't have anything. We can solve both with some placeholder text I think. Most of the time it's going to load quickly, and it would be good to give the user some indication that the data is fresh/valid anyhow.

Maybe a `Refreshing...`, `Loading...`, or `Calculating...` first line and the next two with some non-breaking space. We could also render in disabled style to make it feel really polished `disabledStyle={loading} errorStyle={error}` or something.

Maybe something like this?
`error ? [error, '\xa0', '\xa0'] :  (data || ['Calculating...', '\xa0', '\xa0'])`